### PR TITLE
50 delete during post creation (merge with 47)

### LINKED ISSUE #50: Delete Functionality During Post Creation
## 📝 Task Description 

During the post-creation process, users are able to click the three dots in the upper right-hand corner which prompts them to delete/draft the post. Your task is to implement the delete functionality. This modal should meet the design specifications outlined in the [Figma](https://www.figma.com/file/pxgrEsnOJneqsaWWeEnQoY/FreeBites?type=design&node-id=2904%3A12611&mode=design&t=ASnzrFFtsl0cB5Ux-1).

Screenshot 2024-07-15 at 11 28 15 PM

## ✅ Acceptance Criteria 

- [x] Is the delete confirmation modal styled as per the Figma? (check that its implemented)
- [x] Does the post delete when the user clicks the delete button, and routes them back to the home page?

## 🔍 Subtask Breakdown

1. **Design and Layout**
   - Please ensure that the modals are standardized while delivering this task.
   - ** This means a bonus/extra part of your ticket would be to ensure modals are generally made the same way - try and see if you can make 1 component for all modals **

2. **Functionality**
   - Ensure the post is deleted when they click delete

## 💫 Recommended Approach

Reuse old modal/delete logic!

## ✨ Styling Resources ✨
- [How to CSS](https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web/CSS_basics)
- [Clarence's Styling Presentation](https://docs.google.com/presentation/d/10k0kGCUdR3twBIv9yDJ0hOQZCDFyoy99SGMMBguShFg/edit#slide=id.g24e07eed7b0_1_75)

### PR DESCRIPTION
**Pull request by owennjpr**
# Description

We revamped the modal popup in the post creation window, and added the functionality to the delete button in this modal. We also updated the structure of the modals throughout the app to be more modular and efficient.

## Issues

Resolve #50 

## Test
- log in
- click on the create new post button on the home screen
- take a picture to use for the post
- Continue into the post creation page, however many screens you want to move through
- click the three dots in the top right of the screen, the modal should appear
- press the background or cancel button to close the modal
- press the dots to reopen it
- press the delete button and you should be returned to the home screen with no new post created.

## Possible Downsides

There may still be some references in the code still to the previous implementation of this modal that we missed or failed to remove. Creating new modals will require the use of the FreeBitesModal component from now on as opposed to the react native Modal component.

### CODE REVIEW

File: src/components/home/FreeBitesConfirmationModal.tsx
Comment (johnny-t06) [inline src/components/home/FreeBitesConfirmationModal.tsx:1]: Looks a little different from the Figma

File: src/components/home/FreeBitesModal.tsx
Comment (johnny-t06) [inline src/components/home/FreeBitesModal.tsx:1]: Can we look to align the text on the left (the Cancel should line up as well) and the icons on the right?

**Review by johnny-t06 — CHANGES_REQUESTED** [review]
Good work so far team, I have a few changes requested. Let us know if anything sounds confusing

**Comment by Jet4stream** [comment]
2 things to mention.
- There is a lag between the delete modal and the confirmation delete modal because we need to wait for one modal to close before we open another modal. For this example, we have to wait around 300 milliseconds for the first modal to close.
- The post does not delete when the user presses "yes". Instead, the post is deleted after the user confirms with the confirmation modal. This is because if we delete the post before the confirmation modal, the confirmation modal does not have a chance to show up.

**Pull request merged**